superset/sql_parse.py

```diff
@@ -522,6 +530,43 @@ def _extract_tables_from_statement(
         dialect = SQLGLOT_DIALECTS.get(engine)
         return extract_tables_from_statement(parsed, dialect)
 
+    def is_dml(self) -> bool:
+        """
+        Check if the statement is DML.
+
+        :return: True if the statement is DML
+        """
+        for node in self._parsed.walk():
+            if isinstance(
+                node,
+                (
+                    exp.Insert,
+                    exp.Update,
+                    exp.Delete,
+                    exp.Merge,
+                    exp.Create,
```

Comment:
technically this is DDL not DML, wondering if `is_mutating` would be a better name?

Reply:
@mistercrunch do you think we should also rename the DB modal? It currently says "Allow DML".

Reply:
yes I would. Here's what it says currently ->  <img width="459" alt="Screenshot 2024-09-09 at 9 11 58 AM" src="https://github.com/user-attachments/assets/e200b357-3283-4f01-a053-e9c2112d6aaf">

`Allow DML and DDL` - could be good to unpack the acronyms in the tooltip too. "Allow the execution of DDL (Data Definition Language, ie CREATE, DROP, TRUNCATE, ...) and DML (Data Modification Language, ie INSERT, UPDATE, DELETE, ...) {...}"